Added images and other changes (#13)
uptane/deployment-considerations: exceptional_operations.md

## Rolling back software

Sometimes an OEM may determine that the latest updates are less reliable than previous ones.  In that case, it may be necessary to roll back to a previous update.

By default, Uptane does not allow updates to be rolled back and enforces this action with two mechanisms. First, Uptane rejects any new metadata file with a version number lower than the one contained in the previous metadata file. Second, Uptane will reject any new image associated with a release counter that is lower than the release counter of the previous image in the previous Targets metadata file. The first mechanism prevents an attacker from replaying an old metadata file. The second mechanism prevents an attacker who compromises the Director repository from being able to choose old versions of images, despite being able to sign new metadata. See Figure 1 for an example.

<img align="center" src="assets/images/except_1_.rollback_prev.png" width="500" style="margin: 0px 20px"/>

**Figure 1.** *Uptane prevents rollback attacks by rejecting older: (1) metadata files, and / or (2) images.*

There are at least two ways to allow rollbacks, each with different advantages and disadvantages.

> In the first option, an OEM MAY choose to never increment the release counters of images (see Figure 2). Uptane will accept any new image associated with a release counter, as long as it is equal to the release counter of the previous image in the previous Targets metadata file. If release counters are never incremented, then all images would have the same release counters. In this situation, an ECU would accept the installation of any compatible image referred in the new Targets metadata. (See the [Security pages](https://github.com/uptane/deployment-considerations/blob/master/security_considerations.md) on this website).

Reviewer: ```suggestion
In the first option, an OEM MAY choose to never increment the release counters of images (see Figure 2). Uptane will accept any new image associated with a release counter, as long as it is equal to the release counter of the previous image in the previous Targets metadata file. If release counters are never incremented, then all images would have the same release counters. In this situation, an ECU would accept the installation of any compatible image referred in the new Targets metadata. (See the [security considerations](https://github.com/uptane/deployment-considerations/blob/master/security_considerations.md) document.)
```

In the second option, an OEM increments the release counter of an image whenever it is critical that an ECU not install images with lower release counters. In the example in Figure 3, if an ECU installs “foo.img,” then it cannot install “bar.img.” This is done to prevent the installation of compatible images with lower release counters that have known security vulnerabilities, rather than newer images in which these vulnerabilities have been fixed.

<img align="center" src="assets/images/except_3_diffimage_samecounter2.png" width="500" style="margin: 0px 20px"/>

**Figure 3.** *Uptane forbids the installation of images with lower release counters than what is currently installed.*

The advantage to this method is that it prevents rollback attacks in a situation

> where attackers compromise only the Director repository.  However, there are two disadvantages. First, the release counters for images have to be maintained, even if role B now signs for images previously signed by role A. This is because release counters are always compared to previous Targets metadata files. Second, it is more cumbersome to roll back updates, or deliberately cause ECUs to install older images, because the release counters of these older images are incremented in the new Targets metadata for the Image repository with offline keys. However, this method SHOULD be preferred, because it is more secure. See the [Security pages](https://github.com/uptane/deployment-considerations/blob/master/security_considerations.md) on this website for more techniques that can be used to limit rollback attacks when the Director repository is compromised.

Reviewer: ```suggestion
where attackers compromise only the Director repository.  However, there are two disadvantages. First, the release counters for images have to be maintained, even if role B now signs for images previously signed by role A. This is because release counters are always compared to previous Targets metadata files. Second, it is more cumbersome to roll back updates, or deliberately cause ECUs to install older images, because the release counters of these older images are incremented in the new Targets metadata for the Image repository with offline keys. However, this method SHOULD be preferred, because it is more secure. See the [security considerations](https://github.com/uptane/deployment-considerations/blob/master/security_considerations.md) document for more techniques that can be used to limit rollback attacks when the Director repository is compromised.
```

Due to changes in business relationships, an OEM may need to add or remove a tier-1 supplier from its repositories.

> To add a tier-1 supplier, OEMs SHOULD use the following steps. All three steps should be performed using the guidelines on the [Normal Operating Procedures](https://github.com/uptane/deployment-considerations/blob/master/normal_operation.md) page of this website. First, if the supplier signs its own images, then the OEM SHOULD add a delegation to the supplier on the image repository. Second, the supplier SHOULD deliver metadata and / or images to the OEM. Finally, the OEM SHOULD add the metadata and images to its repositories, possibly test them, and then release them to the affected vehicle. 

Reviewer: ```suggestion
To add a tier-1 supplier, OEMs SHOULD use the following steps. All three steps should be performed using the guidelines in the [Normal Operating Procedures](https://github.com/uptane/deployment-considerations/blob/master/normal_operation.md) document. First, if the supplier signs its own images, then the OEM SHOULD add a delegation to the supplier on the image repository. Second, the supplier SHOULD deliver metadata and / or images to the OEM. Finally, the OEM SHOULD add the metadata and images to its repositories, possibly test them, and then release them to the affected vehicle. 
```
